.profile.d/pgbouncer.sh

```diff
@@ -0,0 +1,283 @@
+export PGBOUNCER_POSTGRESQL_PORT=${PGBOUNCER_POSTGRESQL_PORT:-5432}
+# how long should we wait for connections to go away before shutting down
+export PGBOUNCER_SHUTDOWN_TIMEOUT=${PGBOUNCER_SHUTDOWN_TIMEOUT:-60}
+export PGBOUNCER_STDERR=${PGBOUNCER_STDERR:-"vendor/pgbouncer/stderr.log"}
+
+declare -i PGBOUNCER_SERVICE_PID
+export PGBOUNCER_SERVICE_PID=
+
+declare RED GRN YLW BLU CLR
+
+function is-enabled() {
+  (
+    shopt -s extglob nocasematch
+    [[ "$1" =~ ^(1|true|yes|on)$ ]]
+  )
+}
+
+function define-colors() {
+  if is-enabled "${PGBOUNCER_COLOR_DISABLED}"; then
+    export PUR=''
+    export RED=''
+    export GRN=''
+    export YLW=''
+    export BLU=''
+    export CLR=''
+  else
+    export RED='\033[1;37;41m'
+    export PUR='\033[1;35m'
+    export GRN='\033[0;32m'
+    export BLU='\033[0;34m'
+    export YLW='\033[1;33m'
+    export CLR='\033[0m' # No Color
+  fi
+}
+
+define-colors
+
+function ts() {
+  date '+%Y-%m-%d %T.%3N %Z'
+}
+
+function inf() {
+  echo -e "${PUR}[$(ts)] ${GRN}INFO:  $*${CLR}"
+}
+
+function err() {
+  echo -e "${PUR}[$(ts)] ${RED}ERROR: $*${CLR}"
+}
+
+function at() {
+  inf "${BLU}buildpack=pgbouncer at=$*"
+}
+
+function at-status() {
+  local status="$1"
+  local message="$2"
+
+  case "${status}" in
+  enabled)
+    at "${GRN}ENABLED  ✔︎ ${message}"
+    ;;
+  disabled)
+    at "${YLW}DISABLED ✖︎ ${message}"
+    ;;
+  *)
+    err "${message}"
+    ;;
+  esac
+}
+
+function is-pgbouncer-service-enabled() {
+  if ! is-enabled "${PGBOUNCER_ENABLED}"; then
+    at-status disabled pgbouncer-service
+    return 1
+  else
+    at-status enabled pgbouncer-service
+    return 0
+  fi
+}
+
+function pgbouncer-current-pid() {
+  /bin/ps -eopid,args | grep -E -e "${PGBOUNCER_PROCESS_REGEX}" | awk '{print $1}'
+}
+
+function pgbouncer-get-current-pid() {
+  local current_pid
+  current_pid=$(pgbouncer-current-pid)
+
+  [[ -n ${current_pid} && ${current_pid} -gt 0 ]] && {
+    at "pgbouncer-get-current-pid(): pid=${current_pid}" >&2
+    echo -n "${current_pid}"
+    return 0
+  }
+
+  at "pgbouncer-get-current-pid(): NOT RUNNING" >&2
+  return 1
+}
+
+config-gen() {
+  if [[ -s "${PGBOUNCER_CONFIG_FILE}" ]]; then
+    inf "config-gen(): ${PGBOUNCER_CONFIG_FILE} already exists, skipping"
+    return
+  fi
+
+  # Generate config files
+  at "config-gen() start"
+  source bin/gen-pgbouncer-conf.sh
+  at "config-gen() end"
+
+  # Overwrite config vars with pgbouncer targets
+  export POSTGRES_URLS=${PGBOUNCER_URLS:-DATABASE_URL}
+}
+
+# @description
+#   Returns the number of connections that pgbouncer has established
+#   to PostgresSQL
+function pgbouncer-current-connections-count() {
+  local pid="${1:-$(pgbouncer-get-current-pid)}"
+
+  if command -v lsof >/dev/null; then
+    lsof -p ${pid} -a -P -itcp | grep -c ":${PGBOUNCER_POSTGRESQL_PORT} (ESTABLISHED)"
+  elif command -v netstat >/dev/null; then
+    netstat -an | grep ESTABLISHED | grep -c ":${PGBOUNCER_POSTGRESQL_PORT}"
```

Comment:
NOTE: `PGBOUNCER_POSTGRESQL_PORT` is set to 5432 by default but the buildpack starts it on 6000

Reply:
This is not looking for 6000. It's looking for the remote connection port to the actual DB. Basically the DB pool.

Reply:
Ah gotcha! In that case WDYT of updating the name to distinguish that it's looking for the `PGBOUNCER_PG_SERVER_PORT` or something like that.